frontend/packages/kubevirt-plugin/src/components/vms/types.ts

```diff
@@ -0,0 +1,60 @@
+import { K8sResourceKind, ObjectMetadata } from '@console/internal/module/k8s';
+
+// https://kubevirt.io/api-reference/master/definitions.html#_v1_virtualmachineinstancespec
+export type VMISpec = {
+  affinity: any;
+  dnsConfig: any;
+  dnsPolicy: string;
+  domain?: any;
+  evictionStrategy: any;
+  hostname: string;
+  livenessProbe: any;
+  networks?: any[];
+  nodeSelector: any;
+  readinessProbe: any;
+  subdomain: string;
+  terminationGracePeriodSeconds: number;
+  tolerations: any[];
+  volumes?: any[];
+};
+
+// https://kubevirt.io/api-reference/master/definitions.html#_v1_virtualmachineinstancestatus
+export type VMIStatus = {
+  conditions: any[];
+  interfaces: any[];
+  migrationMethod: string;
+  migrationState: any;
+  nodeName: string;
+  phase: string;
+  reason: string;
+};
+
+export type VMIKind = {
+  spec: VMISpec;
+  status: VMIStatus;
+} & K8sResourceKind;
+
+export type VMITemplate = {
+  metadata?: ObjectMetadata;
+  spec?: VMISpec;
+};
+
+export type VMSpec = {
+  template: VMITemplate;
+  running?: boolean;
+  runStrategy?: any;
+  dataVolumeTemplates?: any[];
+};
+
+export type VMStatus = {
+  conditions?: any[];
+  created?: boolean;
+  ready?: boolean;
+  stateChangeRequests?: any[];
+};
+
+// https://kubevirt.io/api-reference/master/definitions.html#_v1_virtualmachine
+export type VMKind = {
```

Comment:
Honestly, I don't think this is something crucial. Anyway, the rest of recent code-base uses both options where how to place operands.

The one already used in this PR looks more common (statistic retrieved via `grep`).

Reply:
> Honestly, I don't think this is something crucial.

IMO it's more readable to put base object before the extension, instead of the other way around. So my comment was related to better code readability.